dot/network/transaction.go

```diff
@@ -17,7 +17,7 @@ import (
 
 var (
 	_ NotificationsMessage = &TransactionMessage{}
-	_ NotificationsMessage = &transactionHandshake{}
+	_ Handshake            = (*transactionHandshake)(nil)
```

Comment:
why the nil case?

Reply:
This is a check to make sure that transactionHandshake implements Handshake interface. I think, Unlike the previous times, with nil, we end up not allocating any memory.